internal/pkg/crawl/extractor/xml.go

```diff
@@ -19,25 +23,59 @@ func XML(resp *http.Response) (URLs []*url.URL, sitemap bool, err error) {
 		sitemap = true
 	}
 
-	mv, err := mxj.NewMapXml(xmlBody)
+	reader := strings.NewReader(string(xmlBody))
```

Comment:
Should pass `bytes.Reader` instead of `strings.Reader` to ensure the XML decoder works properly on non-UTF-8 encoded bytes.

Reply:
Sorry, I forgot that `strings` are essentially byte slices, which probably won't affect the XML decorder's encoding detection.
However, using `bytes.Reader` directly can avoid `xml.NewDecoder()` creating redundant `bufio.Reader()` for converting strings to bytes.

Reply:
Not an issue.
So XML decoder does internally convert to reader to bytes. Didn't know that. I thought an explicit conversion to string was necessary. Done!